src/Components/App/App.tsx

```diff
@@ -40,21 +41,46 @@ class App extends Component<{}, IAppState> {
       })
   }
 
-  render() {
-    // const allPostsData = this.state.allPosts
-    return (
-      <main>
-        {/* {conditional render} */}
-        {/* <Loading /> */}
+  renderComponent = () => {
+    if (this.state.allPosts.length > 0) {
+      return (
         <section className='main-page'>
           <Searchbar />
           <AllPosts allPosts={this.state.allPosts} />
           <Nav />
         </section>
+      )
+    } else if (this.state.error) {
+      return (
+        <h2>{this.state.error}</h2>
+      )
+    } else {
+      return (
+        <Loading />
+      )
+    }
+  }
 
-        {/* Route here */}
-        {/* <NewPostForm addNewPost={(newPost) => this.addNewPost(newPost)} /> */}
+  render() {
+    return (
+      <main className='app'>
+        <Switch>
+          <Route
+            exact
+            path="/"
+            render={this.renderComponent}
+          />
+          <Route
+            exact
+            path="/new-post"
+            render={() => {
+              <NewPostForm
+                addNewPost={(newPost) => this.addNewPost(newPost)} />
+            }
+            }
```

Comment:
You're passing the function into the component via the router, so it should look something like: `addNewPost={this.addNewPost}`

Reply:
And it needs a return.

Reply:
Return added! Line 78 needs to have newPost passed as a parameter

Reply:
newPost is being passed up from the form, it doesn't have meaning in App.